python/core/auto_generated/qgsrendercontext.sip.in

```diff
@@ -102,7 +102,7 @@ DPI of 88 will be assumed.
 %End
 
 
-    QPainter *painter();
+    QPainter *painter() const;
```

Comment:
Where is this one required? It looks to me that a const QgsRenderContext is being passed somewhere, when they should always be passed as non-const QgsRenderContext& only if painter operations will be made.

Reply:
I have indeed passed a `const QgsRenderContext` in certain places, i.e. [1],  where I just need to query the painter device DPI via `int dpi = context.painter()->device()->logicalDpiX();`, or just to query the context map extent.

[1] https://github.com/kadas-albireo/kadas-albireo2/blob/master/kadas/gui/milx/kadasmilxitem.cpp#L316

Reply:
In that case I'd add a second

    const QPainter* painter() const; 

getter